src/DataProtection/DataProtection/src/LoggingExtensions.cs

```diff
@@ -305,10 +307,15 @@ static LoggingExtensions()
                 eventId: new EventId(39, "WritingDataToFile"),
                 logLevel: LogLevel.Information,
                 formatString: "Writing data to file '{FileName}'.");
-            _readingDataFromRegistryKeyValue = LoggerMessage.Define<RegistryKey, string>(
-                eventId: new EventId(40, "ReadingDataFromRegistryKeyValue"),
-                logLevel: LogLevel.Debug,
-                formatString: "Reading data from registry key '{RegistryKeyName}', value '{Value}'.");
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
```

Comment:
@pranavkm Is there a better way to do this runtime check for a static variable definition?

Reply:
This should be fine. I think DataProtection continues to cross-compile, so you cannot use `OperatingSystem.IsWindows()`